connectors/byoc.py

```diff
@@ -347,131 +387,52 @@ def nested_get(dictionary, keys_, default_=None):
         return nested_get(self.features, keys, default)
 
 
-class Connector:
-    """Represents one doc in `.elastic-connectors` and triggers sync.
-
-    The pattern to use it is:
-
-        await connector.prepare(config)
-        await connector.start_heartbeat(delay)
-        try:
-            await connector.sync(es)
-        finally:
-            await connector.close()
-    """
-
-    def __init__(
-        self,
-        elastic_index,
-        connector_id,
-        doc_source,
-        bulk_options,
-    ):
-
-        self.doc_source = doc_source
-        self.id = connector_id
-        self.index = elastic_index
-        self._update_config(doc_source)
-        self._dirty = False
-        self.client = elastic_index.client
-        self.doc_source["last_seen"] = iso_utc()
-        self._heartbeat_started = self._syncing = False
-        self._closed = False
-        self._start_time = None
-        self._hb = None
-        self.bulk_options = bulk_options
-        self.source_klass = None
-        self.data_provider = None
-
-    def _update_config(self, doc_source):
-        self.status = doc_source["status"]
-        self.sync_now = doc_source.get("sync_now", False)
-        self.native = doc_source.get("is_native", False)
-        self._service_type = doc_source["service_type"]
-        self.index_name = doc_source["index_name"]
-        self._configuration = DataSourceConfiguration(doc_source["configuration"])
-        self.scheduling = doc_source["scheduling"]
-        self.pipeline = PipelineSettings(doc_source.get("pipeline", {}))
-        self._dirty = True
-        self._filtering = Filtering(doc_source.get("filtering", []))
-        self.language_code = doc_source["language"]
-        self.features = Features(doc_source.get("features", {}))
-
+class Connector(ESDocument):
     @property
     def status(self):
-        return self._status
+        return self.get("status")
+
+    @property
+    def service_type(self):
+        return self.get("service_type")
 
-    @status.setter
-    def status(self, value):
-        if isinstance(value, str):
-            value = Status[value.upper()]
-        if not isinstance(value, Status):
-            raise TypeError(value)
+    @property
+    def last_seen(self):
+        last_seen = self.get("last_seen")
+        logger.info(f"last_seen = {last_seen}, type = {last_seen.__class__}")
```

Comment:
shouldn't this be `debug` ?

Reply:
This log shouldn't be here. I added this for my own testing,  because I'm not sure whether the `last_seen` will be parsed as string or time in the response json. I will remove it eventually.